R/install.R

```diff
@@ -396,3 +396,142 @@ build_status_ok <- function(process_log, quiet = FALSE) {
+        } else {
+          message("Writing RTools path to ~/.Renviron ...")
+          write('PATH="${RTOOLS40_HOME}\\usr\\bin;${RTOOLS40_HOME}\\mingw64\\bin;${PATH}"', file = "~/.Renviron", append = TRUE)
+          message("Restart R and run install_cmdstan_toolchain(check_only = TRUE) to confirm the installation was successful.")
+          return(invisible(NULL))
+        }
+      }
+    } else {
+      mingw32_make_path <- dirname(Sys.which("mingw32-make"))
+      gpp_path <- dirname(Sys.which("g++"))
+      # If mingw32-make and g++ are not found
+      # we check typical RTools 3.5 errors and if found fix the path,
+      # otherwise we recommend the user to install RTools 3.5
+      if (!nzchar(mingw32_make_path) || !nzchar(gpp_path)) {
+        rtools_path <- Sys.getenv("RTOOLS35_HOME")
+        if (!nzchar(rtools_path)) {
+          typical_paths <- c(
+            file.path("C:/", "Rtools"),
+            file.path("C:/", "Rtools35"),
+            file.path("C:/", "RTools"),
+            file.path("C:/", "RTools35")
+          )
+          for (p in typical_paths) {
+            if (dir.exists(p)) {
+              rtools_path <- p
+            }
+          }
+        }
+        if (nzchar(rtools_path)) {
+          if(check_only) {
+            error_message <- c(
+              "",
+              "A RTools installation was found, but the PATH is not properly set.",
+              "Run 'cmdstanr::install_cmdstan_toolchain(check_only = FALSE)' to fix the issue."
+            )
+            stop(paste0(error_message, collapse = "\n"), call. = FALSE)
+          }
+          message("Writing RTools path to ~/.Renviron ...")
+          if (nzchar(Sys.getenv("RTOOLS35_HOME"))) {
+            write(paste0('RTOOLS35_HOME=', rtools_path), file = "~/.Renviron", append = TRUE)
+          }
+          write('PATH="${RTOOLS35_HOME}\\bin;${RTOOLS35_HOME}\\mingw_64\\bin;${PATH}"', file = "~/.Renviron", append = TRUE)
+          message("Restart R and run install_cmdstan_toolchain(check_only = TRUE) to confirm the installation was successful.")
+          return(invisible(NULL))
+        } else {
+          error_message <- c(
+            "",
+            "A toolchain was not found. Please install RTools 3.5 and run",
+            "",
+            "write(\'RTOOLS35_HOME=rtools35/install/path/\', path = \"~/.Renviron\", append = TRUE)",
+            "replacing 'rtools35/install/path/' with the actual install path of RTools 3.5.",
+            "",
+            "Then restart R and run 'cmdstanr::install_cmdstan_toolchain(check_only = FALSE)'."
+          )
+          stop(paste0(error_message, collapse = "\n"), call. = FALSE)
+        }
+      }
+    }
+  } else {
+    # On Unix systems we check for make and a suitable compiler
+    make_path <- dirname(Sys.which("make"))
+    if (!nzchar(make_path)) {
+      stop("The 'make' tool was not found. Please install 'make' and restart the R session.", call. = FALSE)
```

Comment:
I thought that `xcode-select --install` was enough but I think at one point @mitzimorris found that she had to install all of xcode. Mitzi am I remembering that correctly? So to be safe we can recommend installing all of xcode I guess. 

To detect if a user is on mac I think this should work:

```r
Sys.info()[["sysname"]] == "Darwin"
```

Reply:
regarding xcode - yes, I saw this during an upgrade to Catalina - under some circumstances this might be necessary - wish I understood which circumstance that is.   I tested a few different things - upgrade, upgrade + migration - on Bob's old 2011 MacBookPro - too many variables there.